File: CPP/map/Divide Array Into Equal Pairs.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) { 
        unordered_map<int, int> mp;
        for(auto num:nums){
            mp[num]++;
        }
        for(auto i:mp){
            if(i.second%2==1)return false;
        }
        return true;
    }
};

File: Python/dictionary/Divide Array Into Equal Pairs.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        thisdict={}
        for num in nums:
            thisdict[num]=thisdict.get(num,0)+1
        for i in thisdict:
            if thisdict[i]%2==1:
                return False
        return True
